### PR TITLE
SingularClosurePartial/TransformsSemigroups

### PR DESCRIPTION
**Pull request by Peter-Ing**

### CODE REVIEW

File: doc/semitrans.xml [hunk 1]
```diff
@@ -376,3 +376,42 @@ gap> Size(WW);
     </Description>
   </ManSection>
 <#/GAPDoc>
+<#GAPDoc Label="SingularClosureTransformationSemigroup">
+   <ManSection>
+    <Oper Name="SingularClosureTransformationSemigroup " Arg="G,n"/>
+    <Returns>A Transformation monoid which contains all singular transformations.</Returns>
```
Comment (MTWhyte) [inline doc/semitrans.xml:382]:
```suggestion
    <Returns>A transformation monoid which contains all singular transformations.</Returns>
```

File: doc/semitrans.xml [hunk 2]
```diff
@@ -376,3 +376,42 @@ gap> Size(WW);
...
+    <Description>
+      When <A>G</A> is a subgroup of the symmertic group on <A>n</A> points then
```
Comment (MTWhyte) [inline doc/semitrans.xml:384]:
```suggestion
      When <A>G</A> is a subgroup of the symmetric group on <A>n</A> points,
```

File: doc/semitrans.xml [hunk 3]
```diff
@@ -376,3 +376,42 @@ gap> Size(WW);
...
   </ManSection>
 <#/GAPDoc>
+<#GAPDoc Label="SingularClosureTransformationSemigroup">
+   <ManSection>
+    <Oper Name="SingularClosureTransformationSemigroup " Arg="G,n"/>
+    <Returns>A Transformation monoid which contains all singular transformations.</Returns>
+    <Description>
+      When <A>G</A> is a subgroup of the symmertic group on <A>n</A> points then 
+	  this returns the least transformation monoid of degree <A>n</A> which contains 
+	  <A>G</A> and all singular transformatiosn.
```
Comment (MTWhyte) [inline doc/semitrans.xml:386]:
```suggestion
	  <A>G</A> and all singular transformations. 
```

File: doc/semitrans.xml [hunk 4]
```diff
@@ -376,3 +376,42 @@ gap> Size(WW);
...
+    </Description>
+  </ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="SingularClosurePartialTransformationSemigroup">
+   <ManSection>
+    <Oper Name="SingularClosurePartialTransformationSemigroup" Arg="G,n"/>
+    <Returns>A partial transformation monoid which contains all singular partial transformations.</Returns>
+    <Description>
+      When <A>G</A> is a subgroup of the symmertic group on <A>n</A> points then
```
Comment (MTWhyte) [inline doc/semitrans.xml:404]:
```suggestion
      When <A>G</A> is a subgroup of the symmetric group on <A>n</A> points,
```

File: doc/semitrans.xml [hunk 5]
```diff
@@ -376,3 +376,42 @@ gap> Size(WW);
...
+  </ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="SingularClosurePartialTransformationSemigroup">
+   <ManSection>
+    <Oper Name="SingularClosurePartialTransformationSemigroup" Arg="G,n"/>
+    <Returns>A partial transformation monoid which contains all singular partial transformations.</Returns>
+    <Description>
+      When <A>G</A> is a subgroup of the symmertic group on <A>n</A> points then 
+	  this returns the least patrial transformation monoid of degree <A>n</A> which
```
Comment (MTWhyte) [inline doc/semitrans.xml:405]:
```suggestion
	  this returns the least partial transformation monoid of degree <A>n</A> which
```

File: doc/semitrans.xml [hunk 6]
```diff
@@ -376,3 +376,42 @@ gap> Size(WW);
...
+	  contains <A>G</A> and all singular partial transformatiosn.
```
Comment (MTWhyte) [inline doc/semitrans.xml:406]:
```suggestion
	  contains <A>G</A> and all singular partial transformations.
```

File: doc/semitrans.xml [hunk 7]
```diff
@@ -376,3 +376,42 @@ gap> Size(WW);
...
+   <ManSection>
+    <Oper Name="SingularClosurePartialTransformationSemigroup" Arg="G,n"/>
+    <Returns>A partial transformation monoid which contains all singular partial transformations.</Returns>
+    <Description>
+      When <A>G</A> is a subgroup of the symmertic group on <A>n</A> points then 
+	  this returns the least patrial transformation monoid of degree <A>n</A> which
+	  contains <A>G</A> and all singular partial transformatiosn. 
+	  
+      <Example><![CDATA[
+	gap> SingularClosurePartialTransformationSemigroup(Group((1,2),(3,4)),4);
```
Comment (MTWhyte) [inline doc/semitrans.xml:409]:
```suggestion
	gap> SingularClosurePartialTransformationSemigroup(Group((1, 2), (3, 4)), 4);
```
(This spacing suggestion is purely based on how gaplint likes lists to be)

File: doc/semitrans.xml [hunk 8]
```diff
@@ -376,3 +376,42 @@ gap> Size(WW);
...
+	<transformation semigroup of degree 5 with 8 generators>
+	gap>Generators(SingularClosurePartialTransformationSemigroup(Group((1,2),(3,4)),4));
```
Comment (MTWhyte) [inline doc/semitrans.xml:411]:
```suggestion
	gap> Generators(SingularClosurePartialTransformationSemigroup(Group((1, 2), (3, 4)), 4));
```
(I think this maybe would need to happen over a couple of lines)

**Comment by james-d-mitchell** [comment]
I tried in 2022 to fix this PR up so that it could be merged, but basically gave up. I don't foresee having the time to do this in the near future, and it might be quicker to do it again from scratch since the original author is no longer available.

**Pull request closed without merging**